### ISSUE TITLE
[Doc] Improve opensearch.client documentation

### ISSUE THREAD

**Issue by Plutone11011** [opensearch-project/opensearch-py#718]
Related to #380

Opensearch client methods `create`, `index`, `delete `could benefit from:

- updated parameters description in documentation, as well as return value and error handling, if relevant
- return values and error handling in user guide

I can contribute to this issue

**Comment by rbpasker** [comment]
+1 on the 'error handling'

it seems that HTTP responses (eg `GET /<index>/_search` AKA `response = client.search(index="<index>")` on a non-existing index) are thrown as errors rather than returned in the `response`, especially since `response` is the nominal return variable for `requests`, and one would check `response['status']'. 

This behavior should be clearly explained in the user guide

https://opensearch-project.github.io/opensearch-py/api-ref/exceptions.html

**Comment by dblock** [comment]
We could use a detailed error handling guide in https://github.com/opensearch-project/opensearch-py/tree/main/guides.

**Comment by rbpasker** [comment]
its actually worse because there's no semantic error checking, eg

`*** Error: RequestError(400, 'resource_already_exists_exception', 'index [opinions/FduukSQQQcuRcp4ZK0w51g] already exists')
`
requires:

```
except TransportError as e:
    if e.status_code == 400 and e.error == 'resource_already_exists_exception':
        # Handle the specific exception
        print("Resource already exists.")
    else:
        # Handle other TransportError exceptions
        print(f"TransportError occurred: {e}")

```




https://opensearch-project.github.io/opensearch-py/api-ref/exceptions.html#opensearchpy.TransportError

**Comment by dblock** [comment]
You're saying that `resource_already_exists_exception` should be a specialized `ResourceAlreadyExistsError`? That would make sense, it's a feature request (please open?).

**Comment by rbpasker** [comment]
yep, there are probably also other errors that I haven't encountered yet that could probably benefit from specializaiton
